hypothesis-python/src/hypothesis/core.py

```diff
@@ -1063,7 +1062,7 @@ def _execute_once_for_engine(self, data: ConjectureData) -> None:
             # OK to re-raise it.
             raise
         except (
-            HypothesisDeprecationWarning,
+            HypothesisWarning,
```

Comment:
Hmm.  I wonder if it would be better to just remove this, so that all `HypothesisWarnings` would be treated as an ordinary test failure in the next clause down?  

If you don't see a substantial downside to that, let's try it out - I'm guessing that we originally treated the deprecation warning as a hard failure just to save on the time spent shrinking and never thought it through again.

Reply:
The problem is (was) that all we saw in that case was a `Flaky` with a `StopTest` cause; no trace of the original `HypothesisWarning`.

The latest revision fixes this, and then the line can be removed.

Side note: `Flaky` can be caused by one *or two different* failures; do you think it would make sense to change it into an `ExceptionGroup` for better reporting?

Reply:
Ooh, yeah, defining it as `class Flaky(BaseExceptionGroup, HypothesisException)` would be lovely.

Reply:
Nice! To clarify, I'm not planning `ExceptionGroup` work in this PR, but I recorded it in a new issue to remember.

Reply:
CI will lead to some minor refactoring, so please hold review until CI is green in ~an hour or so